src/main/java/Duke.java

```diff
@@ -1,163 +1,48 @@
-                        String date = dateObj.format(dateTimeFormat);
-                        task = new Deadline(userWords[0].trim(), date);
-                        list.add(task);
-                        System.out.println(lineSpace + "Got it. I've added this task:\n" + task
-                                + "\nNow you have " + list.size() + " tasks in the list.\n" + lineSpace);
-                        saveData(filePath, list);
-                        break;
-                    case "event":
-                        taskName = sc.nextLine().trim();
-                        if (taskName.isEmpty()) {
-                            throw new DukeException("☹ OOPS!!! The description of an event cannot be empty.");
-                        }
-                        userWords = taskName.split("/at");
-                        if (userWords.length == 1) {
-                            throw new DukeException("☹ OOPS!!! The date/time of an event cannot be empty or is wrongly typed.");
-                        }
-                        task = new Event(userWords[0].trim(), userWords[1].trim());
-                        list.add(task);
-                        System.out.println(lineSpace + "Got it. I've added this task:\n" + task
-                                + "\nNow you have " + list.size() + " tasks in the list.\n" + lineSpace);
-                        saveData(filePath, list);
-                        break;
-                    case "delete":
-                        int deletion = sc.nextInt();
-                        if(deletion < 0 || deletion > list.size()){
-                            throw new DukeException("Task not found");
-                        }
-                        Task temp = list.get(deletion-1);
-                        list.remove(deletion-1);
-                        System.out.println(lineSpace + "Noted. I've removed this task:\n" + temp
-                                + "\nNow you have " + list.size() + " tasks in the list.\n" + lineSpace);
-                        saveData(filePath, list);
-                        break;
-                    case "done":
-                        int taskNo = sc.nextInt();
-                        if(taskNo < 0 || taskNo > list.size()){
-                            throw new DukeException("Task not found");
-                        }
-                        list.get(taskNo - 1).markAsDone();
-                        System.out.println(lineSpace + "Nice! I've marked this task as done:\n"
-                                + list.get(taskNo - 1) + "\n" + lineSpace);
-                        saveData(filePath, list);
-                        break;
-                    default:
-                        throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
-                }
-            } catch (Exception e) {
-                System.out.println(e.getMessage());
+                String fullCommand = ui.readCommand();
```

Comment:
Perhaps a possible alternative implementation is to get the Parser to parse the entire input, not just the command, and allow the Parser to decide which are the parameters to pass on to the various Command objects. Hence, the different Command objects have constructors that take in the remaining parameters (such as the date and time for the DeadlineTask). This way, there is no need to make further UI calls within the Command objects, and the Command object can instead be used solely for decision-making logic. Perhaps we can debate over the merits of either approach :)

Reply:
Your alternative implementation seems cleaner. Mine was a hack-ish way of getting the data part of the input since the Scanner object was inside the UI class.